api/v1/controllers/EventController.js

```diff
@@ -0,0 +1,150 @@
+                        next();
+                        return null;
+                });
+}
+
+function createLocation(req, res, next) {
+        return services.EventService.createLocation(req.body.name,
+                                                    req.body.shortName,
+                                                    req.body.latitude,
+                                                    req.body.longitude)
+                .then(function (result) {
+                        res.body = result.length == 0 ? [] : result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function getLocations(req, res, next) {
+        return services.EventService.getLocations()
+                .then(function (result) {
+                        res.body = result.length == 0 ? [] : result.toJSON();
+                        next();
+                        return null;
+                });
+}
+
+function addLocationToEvents(req, res, next) {
+        return services.EventService.addLocationToEvents(req.body.eventIds,
+                                                         req.body.locationId)
+                .then(function (result) {
+                        return services.EventService.findEventById(req.body.eventId);
+                })
+                .then(function (event) {
+                        res.body = event.toJSON();
+                        next();
+                        return null;
+                })
+                .catch(function (err) {
+                        next(err);
+                        return null;
+                });
+}
+
+function addEventToLocations(req, res, next) {
+        return services.EventService.addEventToLocations(req.body.eventId,
+                                                         req.body.locationIds)
+                .then(function (result) {
+                        return services.EventService.findEventById(req.body.eventId);
+                })
+                .then(function (event) {
+                        res.body = event.toJSON();
+                        next();
+                        return null;
+                })
+                .catch(function (err) {
+                        next(err);
+                        return null;
+                });
+}
+
+function deleteReference(req, res, next) {
+        return services.EventService.deleteReference(req.body.eventIds, req.body.locationIds)
+                .then(function (result) {
+                        next();
+                        return null;
+                });
+}
+
+function deleteEvent(req, res, next) {
+        return services.EventService.deleteEvents(req.body.eventId)
+                .then(function (result) {
+                        next();
+                        return null;
+                });
+}
+
+router.use(bodyParser.json());
+
+router.get('/', getEvents);
+router.get('/locations', getLocations);
+// change me to the non helper function verison
```

Comment:
It may may make more sense to use `router.use(middleware.auth);` here instead of listing it in each endpoint